src/NLog/LogFactory.cs

```diff
@@ -185,6 +185,28 @@ public static IAppDomain CurrentAppDomain
         /// </summary>
         public bool KeepVariablesOnReload { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to automatically call <see cref="LogFactory.Shutdown"/>
+        /// on AppDomain.Unload or AppDomain.ProcessExit
+        /// </summary>
+        public bool AutoShutdown
+        {
+            get { return _autoShutdown; }
+            set
+            {
+                if (value != _autoShutdown)
+                {
+                    _autoShutdown = value;
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__ && !NETSTANDARD1_3
+                    LoggerShutdown -= OnStopLogging;
+                    if (value)
+                        LoggerShutdown += OnStopLogging;
```

Comment:
@snakefoot should we assert this in the tests? I think this is the most important lines to assert:

```
                    LoggerShutdown -= OnStopLogging;
                    if (value)
                        LoggerShutdown += OnStopLogging;
```

Reply:
Did not see such tests for https://github.com/NLog/NLog.Extensions.Logging/pull/314 :)

But I have now created a test, that fails when autoshutdown=false is not working.